aikg/utils/io.py

```diff
@@ -31,3 +34,28 @@ def download_file(url: str, output_path: str | Path):
         for chunk in tqdm(response.iter_content(chunk_size=8192)):
             if chunk:
                 f.write(chunk)
+
+
+def get_sparql_examples(input_path: str | Path) -> List[Document]:
+    """
+    Load SPARQL example query files with first line being the question (starting with #)
+    and the remaining lines being the query. We reformat this content into a document
+    where the page content is the question and the query is attached as metadata
+    """
+    # loop through example files from the input and create a doc with all examples
+    examples_docs = []
+    for file_name in os.listdir(input_path):
+        file_path = os.path.join(input_path, file_name)
+        with open(file_path) as f:
+            examples_temp = []
```

Comment:
**suggestion:** as mentioned above, this whole "file management" part could be moved to prefect workflow, to keep this function more reusable